docs/FAQ.md

```diff
@@ -288,6 +289,38 @@ For a refresher on Go's recommended workspace organization, see the ["How To Wri
 `dep init -gopath`, which falls back to network mode when the packages are not
 found in `GOPATH`. `dep ensure` doesn't work with projects in `GOPATH`.
 
+## How do I use `dep` with Docker?
+
+`dep ensure -vendor-only` creates the vendor folder from a valid `Gopkg.toml` and `Gopkg.lock` without checking for Go code.
+This is especially useful for builds inside docker utilizing cache layers.
+
+Sample dockerfile:
+
+    FROM golang:1.8 AS builder
+
+    RUN apt-get update && apt-get install -y unzip --no-install-recommends && \
+        apt-get autoremove -y && apt-get clean -y && \
+        wget -O dep.zip https://github.com/golang/dep/releases/download/v0.3.0/dep-linux-amd64.zip && \
+        echo '96c191251164b1404332793fb7d1e5d8de2641706b128bf8d65772363758f364  dep.zip' | sha256sum -c - && \
+        unzip -d /usr/bin dep.zip && rm dep.zip
+
+    RUN mkdir -p /go/src/github.com/***
+    WORKDIR /go/src/github.com/***
+
+    COPY Gopkg.toml Gopkg.lock ./
+
+    RUN dep ensure -vendor-only
```

Comment:
Let's add a comment that explains that at this point, `vendor/` has been cached.

Reply:
Added more comments to explain